bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/CloudBigtableScanConfiguration.java

```diff
@@ -98,12 +101,13 @@ public Builder withRequest(ReadRowsRequest request) {
     Builder withKeys(byte[] startKey, byte[] stopKey) {
       final ByteString start = ByteStringer.wrap(startKey);
       final ByteString stop = ByteStringer.wrap(stopKey);
-      request =
-          request.toBuilder()
+      ReadRowsRequest request = this.request == null
+          ? ReadRowsRequest.getDefaultInstance()
+          : this.request.get();
+      return withRequest(staticProvider(request.toBuilder()
               .setRows(RowSet.newBuilder().addRowRanges(
                 RowRange.newBuilder().setStartKeyClosed(start).setEndKeyOpen(stop).build()))
-              .build();
-      return this;
+              .build()));
```

Comment:
This needs some doc explaining that this can only be used when dataflow templates are not used